### PR TITLE
Set timeout for upload tests

### PR DESCRIPTION
**Pull request by lorenzoPrimi**

### CODE REVIEW

File: ooniprobe/View/TestResults/Footer/UploadFooterViewController.m [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #import "TestUtility.h"
 #import "MBProgressHUD.h"
 #import "VersionUtility.h"
+#define STANDARD_TIMEOUT 30
```
Comment (bassosimone) [inline ooniprobe/View/TestResults/Footer/UploadFooterViewController.m]: I would call this minimum timeout

File: ooniprobe/View/TestResults/Footer/UploadFooterViewController.m [hunk 2]
```diff
@@ -107,7 +108,11 @@ -(void)uploadMeasurements:(NSArray *)notUploaded startAt:(NSUInteger)idx{
 
 -(BOOL)uploadMeasurement:(Measurement*)measurement{
     NSString *content = [TestUtility getUTF8FileContent:[measurement getReportFile]];
+    NSUInteger bytes = [content lengthOfBytesUsingEncoding:NSUTF8StringEncoding];
     MKCollectorResubmitSettings *settings = [[MKCollectorResubmitSettings alloc] init];
+    //Timeout dependent on the body size considering a minimum upload speed of 12.5 kbit/s + 10 s
+    NSUInteger timeout = bytes * 8 / (12.5 * 1000) + 10;
+    [settings setTimeout:timeout < STANDARD_TIMEOUT ? STANDARD_TIMEOUT : timeout];
```
Comment (bassosimone) [inline ooniprobe/View/TestResults/Footer/UploadFooterViewController.m]: Strictly speaking we should check that bytes * 8 does not overflow. It would be better to avoid multiplying by `8` and represent the speed in `kbit/s` so we cannot overflow.

Comment (bassosimone) [reply ooniprobe/View/TestResults/Footer/UploadFooterViewController.m]: I would like this functionality to be a static method and I would like to see tests for some corner cases plus a common case.

File: ooniprobe/Utility/TestUtility.mm [hunk 3]
```diff
@@ -143,4 +143,11 @@ + (void)writeString:(NSString*)str toFile:(NSString*)fileName{
         [data writeToFile:fileName atomically:YES];
     }
 }
+
++ (NSUInteger)makeTimeout:(NSUInteger)bytes{
+    //Timeout dependent on the body size considering a minimum upload speed of 12.5 kbit/s + 10 s
```
Comment (bassosimone) [inline ooniprobe/Utility/TestUtility.mm]: I think this comment is now inaccurate because we rounded. I think that's 16 Kbit/s.
